hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/config/ConsumerSenderConfiguration.java

```diff
@@ -0,0 +1,175 @@
+import pl.allegro.tech.hermes.consumers.consumer.sender.http.EmptyHttpHeadersProvidersFactory;
+import pl.allegro.tech.hermes.consumers.consumer.sender.http.Http2ClientHolder;
+import pl.allegro.tech.hermes.consumers.consumer.sender.http.HttpClientsFactory;
+import pl.allegro.tech.hermes.consumers.consumer.sender.http.HttpClientsWorkloadReporter;
+import pl.allegro.tech.hermes.consumers.consumer.sender.http.HttpHeadersProvidersFactory;
+import pl.allegro.tech.hermes.consumers.consumer.sender.http.HttpRequestFactoryProvider;
+import pl.allegro.tech.hermes.consumers.consumer.sender.http.JettyHttpMessageSenderProvider;
+import pl.allegro.tech.hermes.consumers.consumer.sender.http.SendingResultHandlers;
+import pl.allegro.tech.hermes.consumers.consumer.sender.http.SslContextFactoryProvider;
+import pl.allegro.tech.hermes.consumers.consumer.sender.http.auth.HttpAuthorizationProviderFactory;
+import pl.allegro.tech.hermes.consumers.consumer.sender.jms.JmsHornetQMessageSenderProvider;
+import pl.allegro.tech.hermes.consumers.consumer.sender.jms.JmsMetadataAppender;
+import pl.allegro.tech.hermes.consumers.consumer.sender.resolver.EndpointAddressResolver;
+import pl.allegro.tech.hermes.consumers.consumer.sender.resolver.InterpolatingEndpointAddressResolver;
+import pl.allegro.tech.hermes.consumers.consumer.sender.timeout.FutureAsyncTimeout;
+import pl.allegro.tech.hermes.consumers.consumer.trace.MetadataAppender;
+
+import javax.inject.Named;
+import javax.jms.Message;
+import java.util.Optional;
+import java.util.concurrent.ScheduledExecutorService;
+
+import static pl.allegro.tech.hermes.common.config.Configs.CONSUMER_HTTP2_ENABLED;
+import static pl.allegro.tech.hermes.common.config.Configs.CONSUMER_SENDER_ASYNC_TIMEOUT_THREAD_POOL_MONITORING;
+import static pl.allegro.tech.hermes.common.config.Configs.CONSUMER_SENDER_ASYNC_TIMEOUT_THREAD_POOL_SIZE;
+
+@Configuration
+public class ConsumerSenderConfiguration {
+
+    @Bean
+    public MessageBatchSenderFactory httpMessageBatchSenderFactory(ConfigFactory configFactory,
+                                                                   SendingResultHandlers resultHandlers) {
+        return new HttpMessageBatchSenderFactory(configFactory, resultHandlers);
+    }
+
+    @Bean
+    public MessageSenderFactory messageSenderFactory() {
+        return new MessageSenderFactory();
+    }
+
+    @Bean
+    public MessageSenderProviders messageSenderProviders(ProtocolMessageSenderProvider defaultHttpMessageSenderProvider,
+                                                         ProtocolMessageSenderProvider defaultHttpsMessageSenderProvider,
+                                                         ProtocolMessageSenderProvider defaultJmsMessageSenderProvider) {
+        return new MessageSenderProviders(
+                defaultHttpMessageSenderProvider,
+                defaultHttpsMessageSenderProvider,
+                defaultJmsMessageSenderProvider);
+    }
```

Comment:
Ok, but do we want to keep the possibility of splitting the `http` and `https` (it would require adding one more argument to messageSenderFactory) or for now just not overcomplicate things and do as you suggest?